unicorn_mode/samples/speedtest/rust/src/main.rs

```diff
@@ -87,8 +88,40 @@ fn main() {
     fuzz(input_file).unwrap();
 }
 
+struct FuzzCB;
+
+impl<'a, D> UnicornAflExecutorHook<'a, D> for FuzzCB {
+    fn place_input(
+        &mut self,
+        uc: &mut Unicorn<'a, unicornafl::executor::UnicornFuzzData<D>>,
+        afl_input: &[u8],
+        _persistent_round: u64,
+    ) -> bool {
+        // apply constraints to the mutated input
+        if afl_input.len() > INPUT_MAX as usize {
+            //println!("Skipping testcase with leng {}", afl_input.len());
+            return false;
+        }
+
+        uc.mem_write(INPUT_ADDRESS, afl_input).unwrap();
+        uc.mem_write(INPUT_ADDRESS + afl_input.len() as u64, b"\0")
+            .unwrap();
+        true
+    }
+
+    fn validate_crash(
+        &mut self,
+        _uc: &mut Unicorn<'a, unicornafl::executor::UnicornFuzzData<D>>,
+        unicorn_result: Result<(), uc_error>,
+        _input: &[u8],
+        _persistent_round: u64,
+    ) -> bool {
+        unicorn_result.is_err()
+    }
+}
+
 fn fuzz(input_file: &str) -> Result<(), uc_error> {
-    let mut uc = Unicorn::new(Arch::X86, Mode::MODE_64)?;
+    let mut uc = Unicorn::new_with_data(Arch::X86, Mode::MODE_64, UnicornFuzzData::default())?;
```

Comment:
Why not make `new` have the default data and leave the API as is here?

Reply:
Yeah. But we will move to unicorn-engine-sys once it is published.